File: backend/src/db/dbConnect.js
import mongoose from "mongoose";
import { dbName } from "../constants.js";

export const dbConnect = async () => {
    try {
        const connectionInstance = await mongoose.connect(
            `${process.env.MONGODB_URI}/${process.env.dbName}`
        );
        console.log(
            `Database connected successfully at host: ${connectionInstance.connection.host}`
        );
    } catch (error) {
        console.log("Error connecting to database.", error);
        throw error;
    }
};

File: backendFlask/src/db/db_connect.py
from flask_pymongo import PyMongo
from constants import DB_RETRIES
import time

mongo = PyMongo()


def init_db(app):
    retries = DB_RETRIES
    while retries > 0:
        try:
            print("Trying to connect to database")
            # The connection is now handled by Flask-PyMongo
            print(f"Database connected successfully at: {app.config['MONGO_URI']}")
            return
        except Exception as error:
            print("Error connecting to database:", error)
            if retries == 1:
                raise
            retries -= 1
            time.sleep(1)
